base/essentials.jl

```diff
@@ -158,6 +158,12 @@ macro boundscheck(blk)
       $(Expr(:boundscheck,:pop)))
 end
 
+"""
+    inbounds(blk)
+
+Eliminates array bounds by checking within expressions.
```

Comment:
No checking will be performed. Please check the bounds check doc http://docs.julialang.org/en/latest/devdocs/boundscheck?highlight=boundscheck

> Consequently, Julia includes an `@inbounds(...)` macro to tell the compiler to skip such bounds checks within the given block.